plugins/pentaho-googledrive-vfs/core/src/main/java/org/pentaho/googledrive/lifecycle/GoogleDrivePluginLifecycleListener.java

```diff
@@ -0,0 +1,74 @@
+/*!
+* Copyright 2010 - 2017 Pentaho Corporation.  All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.pentaho.googledrive.lifecycle;
+
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileSystemManager;
+import org.apache.commons.vfs2.impl.DefaultFileSystemManager;
+
+import org.pentaho.capabilities.impl.DefaultCapabilityManager;
+import org.pentaho.di.core.annotations.KettleLifecyclePlugin;
+import org.pentaho.di.core.lifecycle.KettleLifecycleListener;
+import org.pentaho.di.core.lifecycle.LifecycleException;
+import org.pentaho.di.core.vfs.KettleVFS;
+
+import org.pentaho.googledrive.vfs.GoogleDriveFileObject;
+import org.pentaho.googledrive.vfs.GoogleDriveFileProvider;
+
+import java.io.File;
+import java.util.Arrays;
+
+@KettleLifecyclePlugin( id = "GoogleDrivePluginLifecycleListener", name = "GoogleDrivePluginLifecycleListener" )
+public class GoogleDrivePluginLifecycleListener implements KettleLifecycleListener {
+
+  public void onEnvironmentInit() throws LifecycleException {
+    try {
+      if ( !new File( GoogleDriveFileObject.resolveCredentialsPath() + "/client_secret.json" ).exists() ) {
```

Comment:
As above - seems like it should be a setting in the kettle.properties, not a new side-car file.